GooglePlayInstant/LoadingScreen/LoadingBar.cs

```diff
@@ -0,0 +1,174 @@
+                    loadingScreenRectTransform.position.y -
+                    LoadingBarYAxisPercentage * loadingScreenRectTransform.position.y);
+
+            SetLoadingBarOutline(loadingBarGameObject);
+            SetLoadingBarFill(loadingBarGameObject);
+        }
+
+        // TODO: check for compatibilty with unity 5.6+
+        private static void SetLoadingBarOutline(GameObject loadingBarGameObject)
+        {
+            var loadingBarOutlineGameObject = new GameObject(LoadingBarOutlineGameObjectName);
+            loadingBarOutlineGameObject.transform.SetParent(loadingBarGameObject.transform);
+
+            loadingBarOutlineGameObject.AddComponent<Image>();
+
+            var loadingBarOutlineImage = loadingBarOutlineGameObject.GetComponent<Image>();
+            loadingBarOutlineImage.sprite = AssetDatabase.GetBuiltinExtraResource<Sprite>("UI/Skin/InputFieldBackground.psd");
+
+            loadingBarOutlineImage.type = Image.Type.Sliced;
+            loadingBarOutlineImage.fillCenter = false;
+
+            // Set size of component
+            var loadingBarOutlineRectTransform = loadingBarOutlineGameObject.GetComponent<RectTransform>();
+            var loadingBarRectTransform = loadingBarGameObject.GetComponent<RectTransform>();
+            loadingBarOutlineRectTransform.sizeDelta = loadingBarRectTransform.sizeDelta;
+
+            // Position outline component
+            loadingBarOutlineGameObject.transform.position = loadingBarGameObject.transform.position;
+        }
+
+        private static void SetLoadingBarFill(GameObject loadingBarGameObject)
+        {
+            var loadingBarFillGameObject = new GameObject(LoadingBarFillGameObjectName);
+            loadingBarFillGameObject.transform.SetParent(loadingBarGameObject.transform);
+
+            loadingBarFillGameObject.AddComponent<Image>();
+            var loadingBarFillImage = loadingBarFillGameObject.GetComponent<Image>();
+            loadingBarFillImage.color = Color.green;
+
+            // Set size of component
+            var loadingBarFillRectTransform = loadingBarFillGameObject.GetComponent<RectTransform>();
+
+            var loadingBarRectTransform = loadingBarGameObject.GetComponent<RectTransform>();
+            loadingBarFillRectTransform.sizeDelta = new Vector2(0,
+                loadingBarRectTransform.sizeDelta.y - LoadingBarFillPadding);
+
+            // Position fill component
+            loadingBarFillGameObject.transform.position = loadingBarGameObject.transform.position;
+        }
+#endif
+
+        /// <summary>
+        /// Updates a loading bar by the progress made by an asynchronous operation up to a specific percentage of
+        /// the loading bar. 
+        /// </summary>
+        public static IEnumerator UpdateLoadingBar(AsyncOperation operation, float percentageOfLoadingBar)
+        {
+            var loadingBarRectTransform = GameObject.Find(LoadingBarGameObjectName).GetComponent<RectTransform>();
```

Comment:
Nit: extract GameObject.Find(LoadingBarGameObjectName) as a local variable and reuse here and below

Reply:
If you're talking about these two lines: 

> var loadingBarRectTransform = GameObject.Find(LoadingBarGameObjectName).GetComponent<RectTransform>();
> 
>             var loadingBarFillRectTransform =
>                 GameObject.Find(LoadingBarFillGameObjectName).GetComponent<RectTransform>();

They are two different game objects.